Formula/derby.rb

```diff
@@ -0,0 +1,46 @@
+class Derby < Formula
+  desc "Apache Derby is an embedded relational database running on JVM"
+  homepage "https://db.apache.org/derby/"
+
+  stable do
+    url "https://www.apache.org/dyn/closer.cgi?path=db/derby/db-derby-10.12.1.1/db-derby-10.12.1.1-bin.tar.gz"
+    mirror "https://archive.apache.org/dist/db/derby/db-derby-10.12.1.1/db-derby-10.12.1.1-bin.tar.gz"
+    sha256 "d233b3f26f652226429c3d2f094f129afc3c3bd6435543e5aeb4a256f95085a8"
+
+    depends_on :java => "1.6+"
+  end
+
+  bottle :unneeded
+
+  option "with-docs-and-demo", "install documentation and demo files"
+
+  def install
+    rm_rf Dir["bin/*.bat"]
+    libexec.install %w[ lib bin test ]
+
+    Pathname.glob("#{libexec}/bin/*") do |file|
+      next if file.directory?
+      (bin/file.basename).write_env_script file, Language::Java.overridable_java_home_env
+    end
+
+    doc.install %w[ index.html LICENSE NOTICE RELEASE-NOTES.html KEYS docs javadoc demo ] if build.with? "docs-and-demo"
+  end
+
+  def caveats; <<-EOS.undent
+    Apache Derby must be in your path for hive executable to work.
+    After installation, set $DERBY_INSTALL and $DERBY_HOME in your profile:
+      export DERBY_INSTALL=#{libexec}
+      export DERBY_HOME=#{libexec}
+
+    You may need to set JAVA_HOME:
+      export JAVA_HOME="$(/usr/libexec/java_home)"
+    EOS
+  end
+
+  test do
+    ENV["DERBY_INSTALL"] = libexec
+    ENV["DERBY_HOME"] = libexec
+
+    assert_match /OS name:         Mac OS X/, shell_output("sysinfo")
```

Comment:
sysinfo belongs to derby binaries. but yes this needs to refactored to include derby server mode testing as well.